vpr/src/route/rr_graph.cpp

```diff
@@ -2501,6 +2501,70 @@ static vtr::NdMatrix<std::vector<int>, 4> alloc_and_load_track_to_pin_lookup(vtr
     return track_to_pin_lookup;
 }
 
+/* Prints all the data about node inode to file fp.                    */
+void print_rr_node(FILE* fp, const std::vector<t_rr_node>& L_rr_node, int inode) {
+    std::string info = describe_rr_node(inode);
+    fprintf(fp, "%s\n", info.c_str());
+
+    fprintf(fp, "%d edge(s):", L_rr_node[inode].num_edges());
+    for (int iconn = 0; iconn < L_rr_node[inode].num_edges(); ++iconn)
+        fprintf(fp, " %d", L_rr_node[inode].edge_sink_node(iconn));
+    fprintf(fp, "\n");
+
+    fprintf(fp, "Switch types:");
+    for (int iconn = 0; iconn < L_rr_node[inode].num_edges(); ++iconn)
+        fprintf(fp, " %d", L_rr_node[inode].edge_switch(iconn));
+    fprintf(fp, "\n");
+
+    fprintf(fp, "Capacity: %d\n", L_rr_node[inode].capacity());
+    fprintf(fp, "R: %g  C: %g\n", L_rr_node[inode].R(), L_rr_node[inode].C());
+    fprintf(fp, "Cost_index: %d\n", L_rr_node[inode].cost_index());
+}
+
+/* Writes out al data about node inode to binary in file fp */
+void write_rr_node(FILE* fp, const std::vector<t_rr_node>& L_rr_node, int inode) {
+    const auto& rr_node = L_rr_node[inode];
+    t_rr_type type = rr_node.type();
```

Comment:
Please add a header to the binary file which has some data like;
 - [ ] [Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files) / [File signatures.](https://en.wikipedia.org/wiki/List_of_file_signatures) 
 - [ ] A binary file version format.
 - [ ] The details in the rr_graph.xml header (`<rr_graph tool_name="vpr" tool_version="8.0.0-dev+vpr-7.0.5-6080-gc64882fe2" tool_comment="Generated from arch file /home/tansell/github/SymbiFlow/symbiflow-arch-defs/ice40/build/ice40-top-routing-virt-hx8k/arch.xml">`)